docs/operators/deduper.md

```diff
@@ -0,0 +1,523 @@
+
+As part of configuring the operator for this use case, we need to
+set the following parameters:
+
+-   ***keyExpression*** - This can be set as
+    the primary key which can be used to uniquely identify a Customer.
+    For example, we can set it to “{Name,
+    Email}”
+
+
+The above configuration is sufficient to resolve the use case.
+
+
+Use case - Dedup with Expiry
+=========================================
+
+Motivation
+-----------------------
+
+The Basic Dedup use case is the most straightforward and is
+usually applied when the amount of data to be processed is not huge.
+However, if the incoming data is huge, or even never-ending, it is
+usually not necessary to keep storing all the data. This is because in
+most real world use cases, the duplicates occur only a short distance
+apart. Hence, after a while, it is usually okay to forget the part of
+the history and consider only limited history for identifying
+duplicates, in the interest of efficiency. In other words, we expire
+(ignore) some tuples which are (or were supposed to be) delivered long
+back. Doing so, reduces the load on the Bucket Store which effectively
+deletes part of the history, thus making the whole process more
+efficient. We call this use case, Dedup with
+expiry.
+
+Expiry Key
+-----------------------
+
+The easiest way to understand this use case is to consider
+time as the criteria of expiring
+tuples. Time is a natural expiry
+key and is in line with the concept of expiry. Formally, an expiry field
+is a field in the input tuple which can be used to discard incoming
+tuples as expired. This expiry key
+usually works with another parameter called Expiry Period defined
+next.
+
+Expiry Period
+--------------------------
+
+The expiry period is the value supplied by the user to define the
+extent of history which should be considered while expiring
+tuples.
+
+Use case Details
+-----------------------------
+
+Consider an incoming stream of system logs. The use case requires
+us to identify duplicate log messages and pass on only the unique ones.
+Another relaxation in the use case is that the log messages which are
+older than a day, may not be considered and must be filtered out as
+expired. The expiry must be measured with respect to the time stamp in
+the logs. For example, if the timestamp in the incoming message is
+`30-12-2014 00:00:00` and the
+latest message that the system has encountered had the time stamp
+`31-12-2014 00:00:00`, then the
+incoming message must be considered as expired. However, if the incoming
+message had any timestamp like `30-12-2014
+00:11:00`, it must be accepted into the system and check
+for a possible duplicate.
+
+The expiry facet in the use case above gives us an advantage in
+that we do not have to compare the incoming record with all the data to check if it is a duplicate.
+At the same time, all the
+incoming data need not be stored. Just a day worth of data needs to be
+stored in order to address the above use case.
```

Comment:
need not be stored. Just ... => need not be stored, just a day's worth is adequate.